### PR TITLE
Improvements to ceph-config playbook

### PR DESCRIPTION
**Pull request by priteau**

### CODE REVIEW

File: etc/kayobe/ansible/ceph-config.yml [hunk 1]
```diff
@@ -2,11 +2,11 @@
 - name: Configure Ceph
   any_errors_fatal: True
   gather_facts: True
-  hosts: storage
+  hosts: mons[0]
   vars:
     kayobe_config_path: "{{ lookup('env', 'KAYOBE_CONFIG_PATH') }}"
     cephadm_ceph_pools:
-      - backup
+      - backups
```
Comment (markgoddard) [inline etc/kayobe/ansible/ceph-config.yml:9]: Doh. Copied from kolla-ansible CI, which was also wrong until recently.

File: etc/kayobe/ansible/ceph-config.yml [hunk 2]
```diff
@@ -21,69 +21,72 @@
...
-            name: "{{ kayobe_config_path }}/kolla/config/{{ item.name }}"
-            mode: 0755
-          with_items: "{{ cephadm_kolla_ceph_services }}"
-          delegate_to: localhost
+    - name: Create ceph.conf
+      command:
+        cmd: cephadm shell -- cat /etc/ceph/ceph.conf | tee /etc/ceph/ceph.conf
+      become: True
+      args:
+        creates: /etc/ceph/ceph.conf
```
Comment (markgoddard) [inline etc/kayobe/ansible/ceph-config.yml:52]: This would prevent changes to the file after creation

File: etc/kayobe/ansible/ceph-config.yml [hunk 3]
```diff
@@ -21,69 +21,72 @@
...
-        - name: Ensure required kolla config directories exist
-          file:
-            state: directory
-            name: "{{ kayobe_config_path }}/kolla/config/{{ item.name }}"
-            mode: 0755
-          with_items: "{{ cephadm_kolla_ceph_services }}"
-          delegate_to: localhost
+    - name: Create ceph.conf
+      command:
+        cmd: cephadm shell -- cat /etc/ceph/ceph.conf | tee /etc/ceph/ceph.conf
```
Comment (markgoddard) [inline etc/kayobe/ansible/ceph-config.yml:49]: Why not use fetch?

File: etc/kayobe/ansible/ceph-config.yml [hunk 4]
```diff
@@ -21,69 +21,72 @@
...
 
-        - name: remove tabs in ceph.conf
-          replace:
-            path: "{{ kayobe_config_path }}/kolla/config/{{ item.name }}/ceph.conf"
-            regexp: "^\t"
-          with_items: "{{ cephadm_kolla_ceph_services }}"
-          delegate_to: localhost
+    - name: copy ceph.conf to enabled services
+      fetch:
+        src: "/etc/ceph/ceph.conf"
```
Comment (markgoddard) [inline etc/kayobe/ansible/ceph-config.yml:65]: You have it on localhost now

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: etc/kayobe/ansible/ceph-config.yml (hunk 3 above)
Comment (priteau) [reply etc/kayobe/ansible/ceph-config.yml:49]: I don't think we can use fetch, as the file is not present on the host. I think it is generated on the fly by cephadm?

Comment (markgoddard) [reply etc/kayobe/ansible/ceph-config.yml:49]: Of course, it executes on the remote hosts. Do you need to use the shell module to pipe?

Comment (priteau) [reply etc/kayobe/ansible/ceph-config.yml:49]: I used cephadm again and this time it created `ceph.conf` on each host. I am removing this part.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**